intake/catalog/entry.py

```diff
@@ -125,6 +126,28 @@ def plots(self):
         """List custom associated quick-plots """
         return list(self._metadata.get('plots', {}))
 
+    def _ipython_display_(self):
+        """Display the entry as a rich object in an IPython session."""
+        contents, warning = self._display_content()
+        display({ # noqa: F821
+            'application/json': contents,
+            'text/plain': pretty_describe(contents)
+        }, metadata={
+            'application/json': { 'root': self._name }
```

Comment:
Does this now error on remote entries, then?
A couple of ways to fix this: 
- use `_name` in remote entries throughout
- make sure `.describe` contains a "name" key and don't use attribute at all
- move these basic attributes to `Entry` base class and define only once (and move `describe()` implementation)

Reply:
Happy to take a crack at whichever of these seems most sensible to you.

Reply:
It seems to me that if the base `Entry` class contains the `display` logic, then it should not use attributes specific to subclasses. This would mean moving at least `_name` and `_driver` to `Entry`.

Reply:
As in, option 3 of your list.

Reply:
Fair enough - although the remote ones don't actually have a `plugin` (it get resolved only at open time), but `None` would be a fine default.

Reply:
I'm actually now taking a closer look at number two. It involves a bunch of changes to test outputs, but is otherwise rather straightforward. Thoughts?

Reply:
Certainly seems the simplest change, and we can maybe refactor as number 3 later